nilearn/tests/test_extmath.py

```diff
@@ -6,13 +6,16 @@
 
 import numpy as np
 
+from sklearn.utils import check_random_state
 from nilearn._utils.extmath import fast_abs_percentile, is_spd
 
 
 def test_fast_abs_percentile():
-    data = np.arange(1, 100)
-    for p in range(10, 100, 10):
-        yield nose.tools.assert_equal, fast_abs_percentile(data, p - 1), p
+    rng = check_random_state(1)
+    data = np.arange(10, 100, 10)
+    rng.shuffle(data)
+    for p in data:
```

Comment:
> Those tests existed independently of the index error.

:confused: what I am trying to get get at is to leave the tests as they were, except for the shuffle which I think was a big flaw since it gave us false confidence that fast_abs_percentile was working when it wasn't.